hal/src/lib.rs

```diff
@@ -6,6 +6,9 @@ pub use paste;
 
 pub mod typelevel;
 
+#[cfg(not(feature = "device"))]
+compile_error!("The HAL requires a target device selection via a Cargo feature");
```

Comment:
Please add to the error message: "or if this is a library, disable target device selection by specifying the `device` feature"